app/src/androidTest/java/com/example/antitheft/structure/GPSCommandHandlerTest.java

```diff
@@ -1,6 +1,103 @@
 package com.example.antitheft.structure;
 
+import com.eis.smslibrary.SMSMessage;
+import com.eis.smslibrary.SMSPeer;
+
+import org.junit.Before;
+import org.junit.Test;
+
+/**
+ * @author Francesco Bau'
+ * @since 29/02/2020
+ */
 public class GPSCommandHandlerTest {
 
+    private SMSMessage smsMessage;
+    private SMSPeer smsPeer;
+
+    static final String LOCATION = "";
+    // WARNING: insert a VALIID telephoneNumber, because it sends an SMS to that telephone number.
+    static final String VALID_TELEPHONE_NUMBER = "+393515186755";
+    static final String TEXT_MESSAGE = "";
+
+
+    /**
+     * Initialization of the fields, before testing.
+     *
+     * @see com.eis.smslibrary.SMSPeer
+     * @see com.eis.smslibrary.SMSMessage
+     */
+    @Before
+    public void init() {
+        smsPeer = new SMSPeer(VALID_TELEPHONE_NUMBER);
+        smsMessage = new SMSMessage(smsPeer, TEXT_MESSAGE);
+    }
+
+    /**
+     * Test what happens if there's an attempt to send a null {@link SMSMessage} .
+     * <p>
+     * {@link NullPointerException} is expected to be thrown, because of the @NonNull annotation.
+     *
+     * @see NullPointerException
+     * @see com.eis.smslibrary.SMSMessage
+     * @see GPSCommandHandler#sendMessage(SMSMessage)
+     */
+    @Test(expected = NullPointerException.class)
+    public void sendNullMessage() {
+        new GPSCommandHandler().sendMessage(null);
```

Comment:
You are totally right!
Though the test fails, because getCurrentLocation() can't work if the MainActivity is fully-closed, since it needs an Activity instance to retrieve the last location.

But thank you very much for the tip, I hadn't thought about that!

And don't be worried, every kind of suggestion is useful! ;)

Reply:
You are totally right!
I don't know why I hadn't thought about that...